src/Parallel/ElementRegistration.hpp

```diff
@@ -0,0 +1,80 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <type_traits>
+
+#include "Utilities/ErrorHandling/Error.hpp"
+#include "Utilities/TMPL.hpp"
+
+/// \cond
+namespace db {
+template <typename DbTagList>
+class DataBox;
+}  // namespace db
+namespace Parallel {
+template <typename Metavariables>
+class GlobalCache;
+}  // namespace Parallel
+/// \endcond
+
+namespace Parallel {
+namespace detail {
+template <typename Metavariables, typename ParallelComponent,
+          typename = std::void_t<>>
+struct has_registration_list : std::false_type {};
+
+template <typename Metavariables, typename ParallelComponent>
+struct has_registration_list<
+    Metavariables, ParallelComponent,
+    std::void_t<typename Metavariables::template registration_list<
```

Comment:
note for future: we need to document this registration stuff in the metavariables as well. Also we could make the `registration_list` a `tmpl::map` to avoid the awkward templating on the component.

Reply:
I agree, and will create an issue so we don't forget

Reply:
https://github.com/sxs-collaboration/spectre/issues/3266